ironfish/src/network/peers/peerManager.ts

```diff
@@ -887,10 +888,16 @@ export class PeerManager {
   ): void {
     // If we receive any message other than an Identity message, close the connection
     if (!isIdentify(message)) {
-      this.logger.debug(
-        `Disconnecting from ${peer.displayName} - Sent unexpected message ${message.type} while waiting for identity`,
-      )
-      peer.close()
+      let error = `Disconnecting from ${peer.displayName} - Sent unexpected message ${message.type} while waiting for identity`
+
+      // If we can parse out their old version, lets display that instead
+      const oldVersion = getIdentifyVersion(message)
+      if (oldVersion) {
+        error = `Old peer attempted to connect to us from ${peer.displayName}: ${oldVersion}`
```

Comment:
Should we reorder the checks in `handleWaitingForIdentityMessage` to look at just the version field first? Maybe checking that the message matches the spec exactly is too aggressive, and we should check it field by field. Or we could log the entire message they send